src/components/LinkModal/LinkModal.js

```diff
@@ -0,0 +1,253 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import defaultTool from 'constants/defaultTool';
+import core from 'core';
+import ActionButton from 'components/ActionButton';
+import { Tabs, Tab, TabPanel } from 'components/Tabs';
+import Button from 'components/Button';
+import actions from 'actions';
+import selectors from 'selectors';
+
+import './LinkModal.scss';
+
+const LinkModal = () => {
+  const [isDisabled, isOpen, totalPages, currentPage, tabSelected] = useSelector(state => [
+    selectors.isElementDisabled(state, 'linkModal'),
+    selectors.isElementOpen(state, 'linkModal'),
+    selectors.getTotalPages(state),
+    selectors.getCurrentPage(state),
+    selectors.getSelectedTab(state, 'linkModal'),
+  ]);
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+
+  const urlInput = React.createRef();
+  const pageNumberInput = React.createRef();
+
+  const [dropdownNumbers, setDropdownNumbers] = useState([]);
+  const [url, setURL] = useState('');
+  const [pageNumber, setPageNumber] = useState(1);
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('linkModal'));
+    setURL('');
+    setPageNumber(1);
+    core.setToolMode(defaultTool);
+  };
+
+  const createLink = () => {
+    const linksResults = [];
+
+    const quads = core.getSelectedTextQuads();
+    const selectedAnnotations = core.getSelectedAnnotations();
+
+    const currPageQuads = quads[currentPage - 1];
+
+    if (currPageQuads) {
+      const selectedText = core.getSelectedText();
+      currPageQuads.forEach(quad => {
+        const link = new Annotations.Link();
+        link.PageNumber = currentPage;
```

Comment:
It'd be good if we can remove the duplicate code here. we can potentially have a function that takes the `X`, `Y`, `Width` and `Height` and use that function here and inside the `if` block below.

Reply:
resolved